google_guest_agent/hostname/hostname_linux.go

```diff
@@ -0,0 +1,92 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package hostname
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"syscall"
+)
+
+const (
+	platformHostsFile = "/etc/hosts"
+	newline           = "\n"
+)
+
+func initPlatform(context.Context) {}
+
+var setHostname = func(hostname string) error {
+	if err := syscall.Sethostname([]byte(hostname)); err != nil {
+		return err
+	}
+	if _, err := exec.LookPath("nmcli"); err == nil {
```

Comment:
what if the system is configured with NetworkManager? have you seen how we are interacting with specific network-managers in the `network/manager` package?

Reply:
If the system is configured with network manager that's fine, this will set the hostname via network manager so that it doesn't override the hostname again with what it sees over DHCP on lease renewal.

This behavior (same as rsyslog) is here to maintain the same behavior as the google_set_hostname script.
https://github.com/GoogleCloudPlatform/guest-configs/blob/master/src/usr/bin/google_set_hostname#L59